src/cata_tiles.cpp

```diff
@@ -502,14 +503,49 @@ void cata_tiles::load_tilejson_from_file(const std::string &tileset_dir, std::if
     // offset should be the total number of sprites loaded from every tileset image
     // eliminate any sprite references that are too high to exist
     // also eliminate negative sprite references
+
+    // loop through all tile ids
     for( auto it = tile_ids.begin(); it != tile_ids.end(); ) {
+        // second is the tile_type describing that id
         auto &td = it->second;
-        td.fg.erase(std::remove_if(td.fg.begin(), td.fg.end(),
-                               [&](int i) { return i >= offset || i < 0; }), td.fg.end());
-        td.bg.erase(std::remove_if(td.bg.begin(), td.bg.end(),
-                               [&](int i) { return i >= offset || i < 0; }), td.bg.end());
+        // loop through fg and bg
+        for (auto *g : {&td.fg, &td.bg}) {
```

Comment:
Same here. It probably works fine as it is, but it is much more confusing than a simple function call.